obbi/active-passive-deployment/templates/obbi/instance-1/wso2ob-obbi-deployment.yaml

```diff
@@ -0,0 +1,123 @@
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# ----------------------------------------------------------------------------------------
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "obbi.resource.prefix" . }}-obbi-1-deployment
+  namespace: {{ .Release.Namespace }}
+spec:
+  replicas: {{ .Values.wso2.deployment.ob.bi.replicas }}
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      deployment: {{template "obbi.resource.prefix" . }}-obbi
+      node: {{template "obbi.resource.prefix" . }}-obbi-1
+  template:
+    metadata:
+      annotations:
+      labels:
+        deployment: {{ template "obbi.resource.prefix" . }}-obbi
+        node: {{template "obbi.resource.prefix" . }}-obbi-1
+    spec:
+      initContainers:
+        - name: init-ob-db
+          image: busybox:1.32
+          command: ['sh', '-c', 'echo -e "Checking for the availability of database server deployment"; while ! nc -z {{ .Values.wso2.deployment.ob.config.datasources.rdbms.server.hostname }} {{ .Values.wso2.deployment.ob.config.datasources.rdbms.server.port }}; do sleep 1; printf "-"; done; echo -e "  >> Database server has started";']
+      containers:
+        - name: obbi
+          {{ if .Values.wso2.deployment.ob.bi.dockerRegistry }}
+          image: {{ .Values.wso2.deployment.ob.bi.dockerRegistry }}/{{ .Values.wso2.deployment.ob.bi.imageName }}
+          {{- $tag := .Values.wso2.deployment.ob.bi.imageTag }}
+          {{- if not (eq $tag "") }}{{- printf ":%s" $tag -}}{{- end }}
+          {{ else }}
+          image: docker.wso2.com/{{ .Values.wso2.deployment.ob.bi.imageName }}:{{ .Values.wso2.deployment.ob.bi.imageTag }}
+          {{ end }}
+          securityContext:
+            runAsUser: 802
+          imagePullPolicy: {{ .Values.wso2.deployment.ob.bi.imagePullPolicy }}
+          ports:
+            - containerPort: 9712
+              protocol: TCP
+            - containerPort: 9612
+              protocol: TCP
+            - containerPort: 7712
+              protocol: TCP
+            - containerPort: 7612
+              protocol: TCP
+            - containerPort: 9092
+              protocol: TCP
+            - containerPort: 9444
+              protocol: TCP
+            - containerPort: 9893
+              protocol: TCP
+          livenessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                - nc -z localhost 9444
+            initialDelaySeconds: {{ .Values.wso2.deployment.ob.bi.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.wso2.deployment.ob.bi.livenessProbe.periodSeconds }}
+          readinessProbe:
+            exec:
+              command:
+                - /bin/sh
+                - -c
+                - nc -z localhost 7612
```

Comment:
As discussed in Mail : "Helm charts for OBBI accelerator" , we have done below change, shall we update this too?

We have omit the readiness probe on checking the thrift data publishing port (7612) . This will make the network traffic flow between active and passive nodes.
And regards to publishing only to active node, we configure the data publishing URL as below on OBAM and OBIAM nodes,
`
server_url = "{tcp://wso2ob-obbi-1-service:7612|tcp://wso2ob-obbi-2-service:7612}"
`
This will route the traffic to obbi instance 1 and if that fails then routes to obbi instance 2

Reply:
Done with 6435c165eb25e4308d83444c4290c4be2df91f42